devito/stencilkernel.py

```diff
@@ -116,12 +119,16 @@ def __init__(self, stencils, **kwargs):
         nodes = SubstituteExpression(subs=subs).visit(nodes)
 
         # Apply the Devito Loop Engine for loop optimization and finalize instantiation
-        dle_state = transform(nodes, mode=dle)
+        dle_state = transform(nodes, mode=set_dle_mode(dle, self.compiler),
+                              compiler=self.compiler)
         body = dle_state.nodes
```

Comment:
In what case would that be useful? 

Note that ``parameters`` is the list of the C function parameters (ie, those used for code generation)